packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1149,8 +1149,11 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             const content = JSON.stringify([...this.chunkMap]);
             addBlobToSummary(summaryTree, chunksBlobName, content);
         }
-        const blobsTree = convertToSummaryTree(this.blobManager.snapshot(), false);
-        addTreeToSummary(summaryTree, blobsTreeName, blobsTree);
+        const snapshot = this.blobManager.snapshot();
+        if (snapshot.entries.length) {
```

Comment:
Maybe be more explicit at testing length === 0?
That said, should it be more generic? I.e. some other process that before we upload it to storage going through tree and trimming empty trees? Seems like a bit more long lasting fix :)

Reply:
I will make it more explicit. 

I don't think we should trim the tree automatically,  because the loading side needs to handle if the tree is not there.

I have considered an explicit check to error out when uploading, but that seems too restrictive for now?

Reply:
Loading should not care.
- DDS works with IChannelStorageService / ChannelStorageService which does not provide a way to see if tree exists - one can only list / read blobs.
- Data store can't have empty DDS sub-trees as we need at least a .attributes blob for it
- So really only ContainerRuntime is the only layer that might care about empty trees, and not for data store paths (they have .component).

So I think blobs is the only one that matters :) And if it works, we can enable validation for everything.
Whether it's good long term - same question as before, we either enforce it or remove from requirements.